### ISSUE TITLE
Saved weights as checkpoints are randomised on loading them in the model

### ISSUE THREAD

**Issue by qaimHassan** [tensorflow/tensorflow#46253]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes, i have a custom U-Net model with conv1d and custom loss function for ignoring the ignore labels
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 2021-01-07 16:16:22.597961: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
v2.4.0-rc4-71-g582c8d236cb 2.4.0
- Python version: 3.8.3
- Bazel version (if compiling from source): 3.7.2
- GCC/Compiler version (if compiling from source): 9.3.0
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
The checkpoints (save_weights) after successfully training the model randomise after loading them into model. The accuracy and precision plummet drastically on the same data which showed 99+ % accuracy while training.
**Describe the expected behavior**
The checkpoints must show the same results after loading, as while the training.

**Standalone code to reproduce the issue**
input_layer = l.Input((seq_len, seq_dep), name='seq')
    model_unet_1d = get_unet(input_layer, labels-1, 16, dropout=0.05, batchnorm=True)
    model_unet_1d.compile(optimizer=Adam(), loss=custom_loss, metrics=[custom_metrics])
    
    print("<<<< Model Compiled >>>>")
    
    callbacks = [
    EarlyStopping(patience=10, verbose=1),
    ReduceLROnPlateau(factor=0.1, patience=5, min_lr=0.00001, verbose=1),
    ModelCheckpoint('model_chkp/model-ctc_4tiles_bce_sam.h5', verbose=1, save_best_only=True, save_weights_only=True)
    ]
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by ravikyram** [comment]
@qaimHassan 

Can you please share colab link or simple standalone code with supporting files to reproduce the issue in our environment.It helps us in debugging faster. Thanks!
